src/odemis/dataio/csv.py

```diff
@@ -41,7 +42,7 @@ def export(filename, data):
     raises:
         IOError in case the spectrum does not contain wavelength metadata.
     '''
-    if data.shape[0] > 1 and numpy.prod(data.shape) == data.shape[0]:
+    if data.shape[0] > 1 and numpy.prod(data.shape) == data.shape[0] and (data.metadata.get(model.MD_ACQ_TYPE, None) == "Spectrum"):
```

Comment:
Drop the check on the shape. Just check the acquisition type is AR or Spectrum.
Inside spectrum, you can have a sub if block for handling 1 or 2 dimensions.